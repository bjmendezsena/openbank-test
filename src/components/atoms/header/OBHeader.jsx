import React from "react";
import "./OBHeader.scss";

export const OBHeader = ({ text = "", children, style = {} }) => {
  const display = children || text;

  if (!display) return;
  return (
    <div test-id="test-id-header" style={{ ...style }} className="OBHeader">
      <h1>{display}</h1>
    </div>
  );
};
