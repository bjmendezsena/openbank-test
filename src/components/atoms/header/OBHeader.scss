@import "../../../styles/base.scss";

.OBHeader {
  padding: 10px;
  h1 {
    font-size: 2rem;
  }
  &::after {
    display: flex;
    content: " ";
    width: 8vh;
    border: 1px solid $blue;
    position: relative;
    top: -2vh;
  }
}
