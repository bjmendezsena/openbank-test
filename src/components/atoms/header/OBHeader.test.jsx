import React from "react";
import { OBHeader } from "./OBHeader";

import { render } from "@testing-library/react";

const getWrapper = (props = {}) => {
  return render(<OBHeader {...props} />);
};

describe("Test in Title component", () => {
  it("Should display the component", () => {
    const text = "Title";

    const wrapper = getWrapper({ text });

    const { getByText } = wrapper;

    expect(getByText(text)).toBeInTheDocument();
  });

  it("Should display the component with the correct styles", () => {
    const text = "Title";

    const wrapper = getWrapper({ text }).asFragment();

    expect(wrapper).toMatchSnapshot();
  });
});
