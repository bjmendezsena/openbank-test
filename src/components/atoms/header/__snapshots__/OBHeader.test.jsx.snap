// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in Title component Should display the component with the correct styles 1`] = `
<DocumentFragment>
  <div
    class="OBHeader"
    test-id="test-id-header"
  >
    <h1>
      Title
    </h1>
  </div>
</DocumentFragment>
`;
